catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java

```diff
@@ -193,87 +246,97 @@ protected Object executeWithSubject() throws Exception {
+      throws IOException {
     console.println("Starting metacard export...");
     Instant start = Instant.now();
-    List<ExportItem> exportedItems = doMetacardExport(zipFile, filter);
+    List<ExportItem> exportedItems = doMetacardExport(zipOutputStream, filter);
     if (exportedItems.isEmpty()) {
       console.println("No metacards found to export, exiting.");
-      FileUtils.deleteQuietly(zipFile.getFile());
+      FileUtils.deleteQuietly(outputFile);
       return null;
     }
 
     console.println("Metacards exported in: " + getFormattedDuration(start));
     console.println("Number of metacards exported: " + exportedItems.size());
     console.println();
 
-    SecurityLogger.audit(
-        "Ids of exported metacards and content:\n{}",
-        exportedItems
-            .stream()
-            .map(ExportItem::getId)
-            .distinct()
-            .collect(Collectors.joining(", ", "[", "]")));
+    auditRecords(exportedItems);
 
     console.println("Starting content export...");
     start = Instant.now();
-    List<ExportItem> exportedContentItems = doContentExport(zipFile, exportedItems);
+    List<ExportItem> exportedContentItems = doContentExport(zipOutputStream, exportedItems);
     console.println("Content exported in: " + getFormattedDuration(start));
     console.println("Number of content exported: " + exportedContentItems.size());
-
     console.println();
 
     if (delete) {
       doDelete(exportedItems, exportedContentItems);
     }
 
     if (!unsafe) {
-      SecurityLogger.audit("Signing exported data. file: [{}]", zipFile.getFile().getName());
-      console.println("Signing zip file...");
-      start = Instant.now();
-      jarSigner.signJar(
-          zipFile.getFile(),
-          System.getProperty(SystemBaseUrl.EXTERNAL_HOST),
-          System.getProperty("javax.net.ssl.keyStorePassword"),
-          System.getProperty("javax.net.ssl.keyStore"),
-          System.getProperty("javax.net.ssl.keyStorePassword"));
-      console.println("zip file signed in: " + getFormattedDuration(start));
+      //  close the stream here to allow the jar writer to certify the full zip.
+      //  Try with resources will close the stream if this is not the case.
+      zipOutputStream.close();
+      signJar(outputFile);
     }
 
     console.println("Export complete.");
-    console.println("Exported to: " + zipFile.getFile().getCanonicalPath());
+    console.println("Exported to: " + outputFile.getCanonicalPath());
     return null;
   }
 
-  private File initOutputFile(String output) {
-    String resolvedOutput;
-    File initialOutputFile = new File(output);
-    if (initialOutputFile.isDirectory()) {
-      // If directory was specified, auto generate file name
-      resolvedOutput = Paths.get(initialOutputFile.getPath(), FILE_NAMER.get()).toString();
-    } else {
-      resolvedOutput = output;
+  private void signJar(File outputFile) {
```

Comment:
📘 Wrapping the jarSigner in a privileged block (squashed commits, making this changed explicit)